sjsonnet/src/sjsonnet/Std.scala

```diff
@@ -274,6 +274,40 @@ object Std {
         }
       )
     },
+    builtin("flatMap", "func", "arr"){ (ev, fs, func: Applyer, arr: Val) =>
+      val res: Val = arr match {
+        case a: Val.Arr =>
+          val arrs = Val.Arr(
+            a.value.map{ i =>
+              Val.Lazy(func.apply(i))
+            }
+          )
+          val out = collection.mutable.Buffer.empty[Val.Lazy]
+          for(x <- arrs.value){
+            x.force match{
+              case Val.Null => // do nothing
```

Comment:
From what I can tell, this `case Val.Null =>` check is unnecessary, since jsonnet.org simply errors out if the second param `arr` is an array and `func` returns `null`. Given that, we might be able to implement this simply as 

```scala
case a: Val.Arr => 
  val results = a.value.map(func.appy(i).asInstanceOf[Val.Arr].value)
  val arrResults = results.collect{case a: Val.Arr => a}
  if (arrResults.size != results.size) {
    val arrResults = results.filter{!_.isInstanceOf[Val.Arr]}
    throw new Error.Delegate("flatMap func must return an array, not " + arrResults.mkString(", "))
  }
  Val.Arr(results.flaten)
```